src/main/java/de/tum/in/www1/artemis/service/AttachmentUnitService.java

```diff
@@ -15,17 +15,21 @@
 
 import de.tum.in.www1.artemis.domain.Attachment;
 import de.tum.in.www1.artemis.domain.Lecture;
+import de.tum.in.www1.artemis.domain.enumeration.AttachmentType;
 import de.tum.in.www1.artemis.domain.lecture.AttachmentUnit;
 import de.tum.in.www1.artemis.domain.lecture.Slide;
 import de.tum.in.www1.artemis.repository.AttachmentRepository;
 import de.tum.in.www1.artemis.repository.AttachmentUnitRepository;
 import de.tum.in.www1.artemis.repository.SlideRepository;
+import de.tum.in.www1.artemis.service.connectors.pyris.PyrisWebhookService;
 import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
 
 @Profile(PROFILE_CORE)
 @Service
 public class AttachmentUnitService {
 
+    private final PyrisWebhookService webhookService;
```

Comment:
I'd prefer you keep the name specific. Otherwise it is not easily understandable what this belongs to when simply seeing its usages.
Also as Iris can also be disabled on a profile level this may not exist and therefore needs to be an optional:
```suggestion
    private final Optional<PyrisWebhookService> pyrisWebhookService;
```

This applies to all other usages of it.